src/charm.py

```diff
@@ -1,103 +1,217 @@
         return {
-            "summary": "httpbin layer",
-            "description": "pebble config layer for httpbin",
-            "services": {
-                "httpbin": {
-                    "override": "replace",
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
-                    "startup": "enabled",
-                    "environment": {
-                        "GUNICORN_CMD_ARGS": f"--log-level {self.model.config['log-level']}"
-                    },
-                }
-            },
+            "API_URL": "/api/v1/",
+            "AIRBYTE_VERSION": "0.57.3",
+            "AIRBYTE_EDITION": "community",
+            "AUTO_DETECT_SCHEMA": "true",
+            "DATABASE_URL": db_url,
+            "DATABASE_USER": db_conn["user"],
+            "DATABASE_PASSWORD": db_conn["password"],
+            "DATABASE_DB": db_name,
+            "DATABASE_HOST": host,
+            "DATABASE_PORT": port,
+            "WORKSPACE_ROOT": "/workspace",
+            "CONFIG_ROOT": "/configs",
+            "TEMPORAL_HOST": self.config["temporal-host"],
+            "WORKER_LOGS_STORAGE_TYPE": self.config["storage-type"].value,
+            "WORKER_STATE_STORAGE_TYPE": self.config["storage-type"].value,
+            "STORAGE_TYPE": self.config["storage-type"].value,
+            "STORAGE_BUCKET_ACTIVITY_PAYLOAD": "payload-storage",
+            "STORAGE_BUCKET_LOG": self.config["storage-bucket-log"],
+            "STORAGE_BUCKET_STATE": self.config["storage-bucket-state"],
+            "STORAGE_BUCKET_WORKLOAD_OUTPUT": self.config["storage-bucket-workload-output"],
+            "CONFIGS_DATABASE_MINIMUM_FLYWAY_MIGRATION_VERSION": "0.40.23.002",
+            "JOBS_DATABASE_MINIMUM_FLYWAY_MIGRATION_VERSION": "0.40.26.001",
+            "MICRONAUT_ENVIRONMENTS": "control-plane",
+            "WORKERS_MICRONAUT_ENVIRONMENTS": "control-plane",
+            "CRON_MICRONAUT_ENVIRONMENTS": "control-plane",
+            "INTERNAL_API_HOST": f"localhost:{INTERNAL_API_PORT}",
+            "LOG_LEVEL": self.config["log-level"].value,
+            "MICROMETER_METRICS_ENABLED": "false",
+            "KEYCLOAK_INTERNAL_HOST": "localhost",
+            "KEYCLOAK_DATABASE_URL": db_url + "?currentSchema=keycloak",
+            "WEBAPP_URL": "airbyte-ui-k8s:8080",
+            "WORKER_ENVIRONMENT": "kubernetes",
+            "WORKSPACE_DOCKER_MOUNT": "airbyte_workspace",
+            "SECRET_PERSISTENCE": "TESTING_CONFIG_DB_TABLE",
+            "CONNECTOR_BUILDER_SERVER_API_HOST": f"localhost:{CONNECTOR_BUILDER_SERVER_API_PORT}",
+            "S3_LOG_BUCKET_REGION": "",
+            "MINIO_ENDPOINT": "http://minio:9000",
+            "S3_LOG_BUCKET": "airbyte-dev-logs",
+            "S3_PATH_STYLE_ACCESS": "true",
+            "SHOULD_RUN_NOTIFY_WORKFLOWS": "true",
+            "STATE_STORAGE_MINIO_ACCESS_KEY": "minio",
+            "STATE_STORAGE_MINIO_SECRET_ACCESS_KEY": "minio123",
```

Comment:
question: What are the ramifications of exposing this in plaintext?

There are other secret keys in this dictionary as well.

Reply:
These are just placeholders for now, I've included a TODO above this dict to replace all of these values with values that come from the relations with the PostgreSQL and MinIO charms. This will come in the next PR.

Reply:
Works for me.